CRM/MembersOnlyEvent/Hook/BuildForm/Register.php

```diff
@@ -62,7 +62,14 @@ protected function buildForm($formName, &$form) {
               continue 2;
             }
 
-            $priceFieldValueID = $element->getAttribute('value');
+            if ($isCheckboxElement) {
```

Comment:
I would alter this to : 

```

  if ($isCheckboxElement) {
   // for civicrm versions >= 5.35.0
    $priceFieldValueID = $element->getAttribute('id');
  }
  else {
    // for civicrm versions < 5.35.0
    $priceFieldValueID = $element->getAttribute('value');
  }
```

Reply:
Sorry, I've updated my PR,
This is not CiviCRM 5.35.1 issue, I tried it on 5.28 and it seems that `$element->getAttribute('value')` for checkbox will always return "1"
Hence  we need to use the id attribute.